src/MPAGSCipher/CaesarCipher.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef MPAGSCIPHER_CAESARCIPHER_HPP
+#define MPAGSCIPHER_CAESARCIPHER_HPP
+
+#include "CipherMode.hpp"
+
+#include <string>
+
+class CaesarCipher {
+    public:
+        explicit CaesarCipher (std::size_t key);
+        explicit CaesarCipher (std::string key);
+
+        std::string applyCipher (const std::string inputText, const CipherMode encrypt);
```

Comment:
Better to take `std::string` objects by reference, and add missing `const`-ness:
```suggestion
        explicit CaesarCipher (const std::size_t key);
        explicit CaesarCipher (const std::string& key);

        std::string applyCipher (const std::string& inputText, const CipherMode encrypt);
```